backend/src/main/java/com/example/subjecthub/entity/SubjectComment.java

```diff
@@ -0,0 +1,155 @@
+package com.example.subjecthub.entity;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.util.Date;
+
+@Entity
+@Table(name = "comments")
+@ParametersAreNonnullByDefault
+public class SubjectComment {
+
+    @Id
+    @GeneratedValue
+    @Column(nullable = false, name = "comment_id")
+    private Long id;
+
+    //@JsonIgnore
+    @ManyToOne
+    @JoinColumn(name = "user_id")
+    @JsonIgnoreProperties(value = {"password", "email", "comments"})
+    private SubjectHubUser user;
+
+    //@JsonIgnore
+    @ManyToOne
+    @JoinColumn(name = "subject_id")
+    @JsonIgnoreProperties(value = {"creditPoints", "description", "minRequirements","faculty", "undergrad", "postgrad",
```

Comment:
Could you not just use @JsonIgnore?

Reply:
I wanted a very small number of crucial attributes/keys to show up in the json. This way if it's needed in the future (for example a user-specific comments list, certain details like which subject the comment is talking about show up)

Reply:
also helpful during testing to ensure that the right thing has actually been done. as some basic user and subject info provide context to individual comments